src/HelloWormhole.sol

```diff
@@ -7,59 +7,107 @@ import "wormhole-relayer-sdk/interfaces/IWormholeReceiver.sol";
-            targetChain,
-            0,
-            GAS_LIMIT
+            targetChain, // Wormhole Chain ID
+            0, // value we're sending in the message (added to the returned cost)
+            GAS_LIMIT // The maximum gas we're willing to cover
         );
     }
 
+    /**
+     * @notice Sends the `greeting` message to the target chain and address by invoking the relayer and covering the gas
+     * costs estimated by the relayer.
+     *
+     * @param targetChain in Wormhole Chain ID format
+     * @param targetAddress the address of the contract on the target chain the message should be relayed to
+     * @param greeting the message we'd like to relay
+     */
     function sendCrossChainGreeting(
         uint16 targetChain,
         address targetAddress,
         string memory greeting
     ) public payable {
+        // request a quote for the gas required to deliver the message
+        // on the target chain
         uint256 cost = quoteCrossChainGreeting(targetChain);
 
+        // If the caller did not pass enough value, fail the transaction
+        // otherwise the value will come from the contract's balance
+        require(msg.value >= cost, "Not enough value was passed.");
+
+        // Note: for simplicity, we're skipping a check
+        //  on the size of the greeting passed here
+
+        // call the relayer contract, passing enough value to cover
+        // the cost of gas on delivery of the message
         wormholeRelayer.sendPayloadToEvm{value: cost}(
-            targetChain,
-            targetAddress,
+            targetChain, // Wormhole Chain ID on which to deliver the message
+            targetAddress, // Intended recipient of the message (contract address on target chain)
             abi.encode(greeting), // payload
             0, // no receiver value needed since we're just passing a message
-            GAS_LIMIT
+            GAS_LIMIT // Maximum
         );
 
+        // If the caller passed more value than is used to cover the cross chain call,
+        // refund them the rest of the value balance
         if (msg.value > cost) {
             (bool success, ) = msg.sender.call{value: msg.value - cost}("");
             require(success, "Returning excess funds failed");
         }
     }
 
+    /**
+     * @notice Receives the `greeting` message sent from the source chain. Invoked
+     * by the relayer.
+     *
+     * @param payload The raw payload sent from the origin chain
+     * @param sourceAddress The raw payload sent from the origin chain
+     * @param sourceChain the address of the contract on the target chain the message should be relayed to
+     */
     function receiveWormholeMessages(
         bytes memory payload,
         bytes[] memory, // additionalVaas
         bytes32 sourceAddress,
         uint16 sourceChain,
         bytes32 // deliveryHash
     ) public payable override {
+        // Don't allow any messages from senders besides the one we're expecting
```

Comment:
Specifically, all messages must pass through the WormholeRelayer contract so that invariants can be enforced.